### ISSUE TITLE
How to retrain or resume training SVTRnet efficiently?

### ISSUE THREAD

**Issue by mobassir94** [PaddlePaddle/PaddleOCR#6757]
i trained svtrnet for bangla text recognition task as discussed here : https://github.com/PaddlePaddle/PaddleOCR/issues/6632 ,i trained the model for long time(6-7 days) and then got acc 0.8.
after getting 0.8 i faced technical difficulties and training notebook got killed unfortunately. now i am trying to retrain/resume training like this : 

`python3 tools/train.py -c configs/rec/rec_svtrnet.yml -o Global.pretrained_model=/backup2/synthtiger/bangla/PaddleOCR/output_80/rec/svtr/best_accuracy 
`
when i got 0.8 acc the learning rate was 0.000235 so i also tried this : 

`python3 tools/train.py -c configs/rec/rec_svtrnet.yml -o Global.pretrained_model=/backup2/synthtiger/bangla/PaddleOCR/output_80/rec/svtr/best_accuracy Optimizer.lr.learning_rate=0.000235 `

but during retraining i see the learning rate starts from 0.00000 and ACC starts from around 0.54 like this : 

```
/home/apsisdev/.local/lib/python3.8/site-packages/scipy/fft/__init__.py:97: DeprecationWarning: The module numpy.dual is deprecated.  Instead of using dual, use the functions directly from numpy or scipy.
  from numpy.dual import register_func
/home/apsisdev/.local/lib/python3.8/site-packages/scipy/sparse/sputils.py:17: DeprecationWarning: `np.typeDict` is a deprecated alias for `np.sctypeDict`.
  supported_dtypes = [np.typeDict[x] for x in supported_dtypes]
/home/apsisdev/.local/lib/python3.8/site-packages/scipy/special/orthogonal.py:81: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
  from numpy import (exp, inf, pi, sqrt, floor, sin, cos, around, int,
[2022/07/01 00:30:08] ppocr INFO: Architecture : 
[2022/07/01 00:30:08] ppocr INFO:     Backbone : 
[2022/07/01 00:30:08] ppocr INFO:         depth : [3, 6, 3]
[2022/07/01 00:30:08] ppocr INFO:         embed_dim : [64, 128, 256]
[2022/07/01 00:30:08] ppocr INFO:         img_size : [32, 100]
[2022/07/01 00:30:08] ppocr INFO:         last_stage : True
[2022/07/01 00:30:08] ppocr INFO:         local_mixer : [[7, 11], [7, 11], [7, 11]]
[2022/07/01 00:30:08] ppocr INFO:         mixer : ['Local', 'Local', 'Local', 'Local', 'Local', 'Local', 'Global', 'Global', 'Global', 'Global', 'Global', 'Global']
[2022/07/01 00:30:08] ppocr INFO:         name : SVTRNet
[2022/07/01 00:30:08] ppocr INFO:         num_heads : [2, 4, 8]
[2022/07/01 00:30:08] ppocr INFO:         out_channels : 192
[2022/07/01 00:30:08] ppocr INFO:         out_char_num : 25
[2022/07/01 00:30:08] ppocr INFO:         patch_merging : Conv
[2022/07/01 00:30:08] ppocr INFO:         prenorm : False
[2022/07/01 00:30:08] ppocr INFO:     Head : 
[2022/07/01 00:30:08] ppocr INFO:         name : CTCHead
[2022/07/01 00:30:08] ppocr INFO:     Neck : 
[2022/07/01 00:30:08] ppocr INFO:         encoder_type : reshape
[2022/07/01 00:30:08] ppocr INFO:         name : SequenceEncoder
[2022/07/01 00:30:08] ppocr INFO:     Transform : 
[2022/07/01 00:30:08] ppocr INFO:         name : STN_ON
[2022/07/01 00:30:08] ppocr INFO:         num_control_points : 20
[2022/07/01 00:30:08] ppocr INFO:         stn_activation : none
[2022/07/01 00:30:08] ppocr INFO:         tps_inputsize : [32, 64]
[2022/07/01 00:30:08] ppocr INFO:         tps_margins : [0.05, 0.05]
[2022/07/01 00:30:08] ppocr INFO:         tps_outputsize : [32, 100]
[2022/07/01 00:30:08] ppocr INFO:     algorithm : SVTR
[2022/07/01 00:30:08] ppocr INFO:     model_type : rec
[2022/07/01 00:30:08] ppocr INFO: Eval : 
[2022/07/01 00:30:08] ppocr INFO:     dataset : 
[2022/07/01 00:30:08] ppocr INFO:         data_dir : /backup2/synthtiger/bangla/PaddleOCR/horizontal_valid/
[2022/07/01 00:30:08] ppocr INFO:         label_file_list : ['/backup2/synthtiger/bangla/PaddleOCR/horizontal_valid/gt.txt']
[2022/07/01 00:30:08] ppocr INFO:         name : SimpleDataSet
[2022/07/01 00:30:08] ppocr INFO:         transforms : 
[2022/07/01 00:30:08] ppocr INFO:             DecodeImage : 
[2022/07/01 00:30:08] ppocr INFO:                 channel_first : False
[2022/07/01 00:30:08] ppocr INFO:                 img_mode : BGR
[2022/07/01 00:30:08] ppocr INFO:             CTCLabelEncode : None
[2022/07/01 00:30:08] ppocr INFO:             RecResizeImg : 
[2022/07/01 00:30:08] ppocr INFO:                 character_dict_path : None
[2022/07/01 00:30:08] ppocr INFO:                 image_shape : [3, 64, 256]
[2022/07/01 00:30:08] ppocr INFO:                 padding : False
[2022/07/01 00:30:08] ppocr INFO:             KeepKeys : 
[2022/07/01 00:30:08] ppocr INFO:                 keep_keys : ['image', 'label', 'length']
[2022/07/01 00:30:08] ppocr INFO:     loader : 
[2022/07/01 00:30:08] ppocr INFO:         batch_size_per_card : 2048
[2022/07/01 00:30:08] ppocr INFO:         drop_last : False
[2022/07/01 00:30:08] ppocr INFO:         num_workers : 8
[2022/07/01 00:30:08] ppocr INFO:         shuffle : False
[2022/07/01 00:30:08] ppocr INFO: Global : 
[2022/07/01 00:30:08] ppocr INFO:     cal_metric_during_train : True
[2022/07/01 00:30:08] ppocr INFO:     character_dict_path : /backup2/synthtiger/bangla/PaddleOCR/ppocr/utils/bn_char.txt
[2022/07/01 00:30:08] ppocr INFO:     checkpoints : None
[2022/07/01 00:30:08] ppocr INFO:     distributed : False
[2022/07/01 00:30:08] ppocr INFO:     epoch_num : 100
[2022/07/01 00:30:08] ppocr INFO:     eval_batch_step : [0, 2000]
[2022/07/01 00:30:08] ppocr INFO:     infer_img : doc/imgs_words_en/41.jpg
[2022/07/01 00:30:08] ppocr INFO:     infer_mode : False
[2022/07/01 00:30:08] ppocr INFO:     log_smooth_window : 20
[2022/07/01 00:30:08] ppocr INFO:     max_text_length : 25
[2022/07/01 00:30:08] ppocr INFO:     pretrained_model : /backup2/synthtiger/bangla/PaddleOCR/output_80/rec/svtr/best_accuracy
[2022/07/01 00:30:08] ppocr INFO:     print_batch_step : 10
[2022/07/01 00:30:08] ppocr INFO:     save_epoch_step : 1
[2022/07/01 00:30:08] ppocr INFO:     save_inference_dir : None
[2022/07/01 00:30:08] ppocr INFO:     save_model_dir : /backup2/synthtiger/bangla/PaddleOCR/output/rec/svtr/
[2022/07/01 00:30:08] ppocr INFO:     save_res_path : /backup2/synthtiger/bangla/PaddleOCR/output/rec/predicts_svtr_tiny.txt
[2022/07/01 00:30:08] ppocr INFO:     use_gpu : True
[2022/07/01 00:30:08] ppocr INFO:     use_space_char : True
[2022/07/01 00:30:08] ppocr INFO:     use_visualdl : False
[2022/07/01 00:30:08] ppocr INFO: Loss : 
[2022/07/01 00:30:08] ppocr INFO:     name : CTCLoss
[2022/07/01 00:30:08] ppocr INFO: Metric : 
[2022/07/01 00:30:08] ppocr INFO:     main_indicator : acc
[2022/07/01 00:30:08] ppocr INFO:     name : RecMetric
[2022/07/01 00:30:08] ppocr INFO: Optimizer : 
[2022/07/01 00:30:08] ppocr INFO:     base_lr : 0.000235
[2022/07/01 00:30:08] ppocr INFO:     beta1 : 0.9
[2022/07/01 00:30:08] ppocr INFO:     beta2 : 0.99
[2022/07/01 00:30:08] ppocr INFO:     epsilon : 8e-08
[2022/07/01 00:30:08] ppocr INFO:     lr : 
[2022/07/01 00:30:08] ppocr INFO:         learning_rate : 0.000235
[2022/07/01 00:30:08] ppocr INFO:         name : Cosine
[2022/07/01 00:30:08] ppocr INFO:         warmup_epoch : 2
[2022/07/01 00:30:08] ppocr INFO:     name : AdamW
[2022/07/01 00:30:08] ppocr INFO:     no_weight_decay_name : norm pos_embed
[2022/07/01 00:30:08] ppocr INFO:     one_dim_param_no_weight_decay : True
[2022/07/01 00:30:08] ppocr INFO:     weight_decay : 0.05
[2022/07/01 00:30:08] ppocr INFO: PostProcess : 
[2022/07/01 00:30:08] ppocr INFO:     name : CTCLabelDecode
[2022/07/01 00:30:08] ppocr INFO: Train : 
[2022/07/01 00:30:08] ppocr INFO:     dataset : 
[2022/07/01 00:30:08] ppocr INFO:         data_dir : /backup2/synthtiger/bangla/PaddleOCR/train_data/
[2022/07/01 00:30:08] ppocr INFO:         label_file_list : ['/backup2/synthtiger/bangla/PaddleOCR/train_data/gt.txt']
[2022/07/01 00:30:08] ppocr INFO:         name : SimpleDataSet
[2022/07/01 00:30:08] ppocr INFO:         transforms : 
[2022/07/01 00:30:08] ppocr INFO:             DecodeImage : 
[2022/07/01 00:30:08] ppocr INFO:                 channel_first : False
[2022/07/01 00:30:08] ppocr INFO:                 img_mode : BGR
[2022/07/01 00:30:08] ppocr INFO:             CTCLabelEncode : None
[2022/07/01 00:30:08] ppocr INFO:             RecResizeImg : 
[2022/07/01 00:30:08] ppocr INFO:                 character_dict_path : None
[2022/07/01 00:30:08] ppocr INFO:                 image_shape : [3, 64, 256]
[2022/07/01 00:30:08] ppocr INFO:                 padding : False
[2022/07/01 00:30:08] ppocr INFO:             KeepKeys : 
[2022/07/01 00:30:08] ppocr INFO:                 keep_keys : ['image', 'label', 'length']
[2022/07/01 00:30:08] ppocr INFO:     loader : 
[2022/07/01 00:30:08] ppocr INFO:         batch_size_per_card : 1280
[2022/07/01 00:30:08] ppocr INFO:         drop_last : True
[2022/07/01 00:30:08] ppocr INFO:         num_workers : 8
[2022/07/01 00:30:08] ppocr INFO:         shuffle : True
[2022/07/01 00:30:08] ppocr INFO: profiler_options : None
[2022/07/01 00:30:08] ppocr INFO: train with paddle 2.3.0 and device Place(gpu:0)
[2022/07/01 00:30:08] ppocr INFO: Initialize indexs of datasets:['/backup2/synthtiger/bangla/PaddleOCR/train_data/gt.txt']
[2022/07/01 00:32:29] ppocr INFO: Initialize indexs of datasets:['/backup2/synthtiger/bangla/PaddleOCR/horizontal_valid/gt.txt']
W0701 00:32:41.625279 2347589 gpu_context.cc:278] Please NOTE: device: 0, GPU Compute Capability: 8.6, Driver API Version: 11.6, Runtime API Version: 11.2
W0701 00:32:41.644074 2347589 gpu_context.cc:306] device: 0, cuDNN Version: 8.1.
[2022/07/01 00:33:20] ppocr INFO: load pretrain successful from /backup2/synthtiger/bangla/PaddleOCR/output_80/rec/svtr/best_accuracy
[2022/07/01 00:33:21] ppocr INFO: train dataloader has 7402 iters
[2022/07/01 00:33:21] ppocr INFO: valid dataloader has 463 iters
[2022/07/01 00:33:21] ppocr INFO: During the training process, after the 0th iteration, an evaluation is run every 2000 iterations
[2022/07/01 00:56:24] ppocr INFO: epoch: [1/100], global_step: 10, lr: 0.000000, acc: 0.537891, norm_edit_dis: 0.876760, loss: 5.682976, avg_reader_cost: 134.79555 s, avg_batch_cost: 138.30644 s, avg_samples: 1280.0, ips: 9.25481 samples/s, eta: 1184 days, 20:57:22
[2022/07/01 01:08:57] ppocr INFO: epoch: [1/100], global_step: 20, lr: 0.000000, acc: 0.540625, norm_edit_dis: 0.878509, loss: 5.502254, avg_reader_cost: 72.45537 s, avg_batch_cost: 75.29117 s, avg_samples: 1280.0, ips: 17.00067 samples/s, eta: 914 days, 22:25:36
[2022/07/01 01:21:28] ppocr INFO: epoch: [1/100], global_step: 30, lr: 0.000000, acc: 0.538281, norm_edit_dis: 0.879901, loss: 5.402974, avg_reader_cost: 69.41810 s, avg_batch_cost: 75.14283 s, avg_samples: 1280.0, ips: 17.03423 samples/s, eta: 824 days, 12:36:41
[2022/07/01 01:31:42] ppocr INFO: epoch: [1/100], global_step: 40, lr: 0.000000, acc: 0.539844, norm_edit_dis: 0.881283, loss: 5.478276, avg_reader_cost: 56.02523 s, avg_batch_cost: 61.31626 s, avg_samples: 1280.0, ips: 20.87538 samples/s, eta: 749 days, 16:54:49

```
now from 0.54 to 0.8 acc increment i will have to wait 2-3 more days as it takes time,i would like to know how can i retrain in the second stage and quickly get close to the accuracy of previously trained best model which is 0.8 in my case?

@WenmuZhou @dyning @LDOUBLEV @tink2123 @MissPenguin @Evezerest @littletomatodonkey @andyjpaddle @weisy11 @D-DanielYang @Topdu @sdcb @ZeyuChen @bingooo

**Comment by andyjpaddle** [comment]
The resume train command is `python3 tools/train.py -c configs/rec/rec_svtrnet.yml -o Global. checkpoints =/backup2/synthtiger/bangla/PaddleOCR/output_80/rec/svtr/latest`

**Comment by mobassir94** [comment]
hi @andyjpaddle 
in svtr directory i have 5 files,
1. best_accuracy.pdopt
2. best accuracy.pdparams
3. best_accuracy.states
4. config.yml
5. train.log

I trained the model for 7 days but no file called "latest" was generated, what can be the reason?

**Comment by andyjpaddle** [comment]
There should be extra three files,` latest.pdopt, latest.pdparams, latest.states`,  make sure you use the latest code. If the best accuracy is obtained from the latest epoch, you can also  use this command `python3 tools/train.py -c configs/rec/rec_svtrnet.yml -o Global. checkpoints =/backup2/synthtiger/bangla/PaddleOCR/output_80/rec/svtr/best_accuracy`.  But 'latest' is recommended.

**Comment by mobassir94** [comment]
@andyjpaddle yes i believe the best accuracy is obtained from the latest epoch,i don't know why other 3 files are missing,i cloned the repository 3-4 weeks ago for setting up environment and installing paddleocr,so i believe i am not using very old paddle ocr version, also i checked best_accuracy in inference mode, according to inference results on few test images looks like the best_accuracy weight was predicting with 0.8 acc and not with 0.5 acc as most of the predictions were correct.
i don't know why during retraining, learning rate starts from 0.0000
this can be one issue for what during retraining ACC starts from 0.54 instead of something around 0.8

**Comment by andyjpaddle** [comment]
Resume train can refer to this doc https://github.com/PaddlePaddle/PaddleOCR/blob/release/2.5/doc/doc_en/recognition_en.md#22-load-trained-model-and-continue-training

**Comment by HSILA** [comment]
@andyjpaddle
While resuming will preserve learning rate, train accuracy and loss, in my opinion it has two issues:
1. It will start from epoch 1. let's say the training stopped in 900/1000, after resuming it will start from 1/1000 not 901/1000.
2. It doesn't know best validation accuracy from previous training and so it will overwrite it, let's say the best val accuracy from previous training was 80%, when you resume training in the first validation (let's say 70%) it will overwrite the previous 80% validation accuracy!!

**Comment by mobassir94** [comment]
@HSILA why  it will start from 1/1000 not 901/1000? i have saved copy version of 80% acc weight files seperately in a folder, let me know if there is any way of resuming training with those weights please without compromising acc,thanks in advance

**Comment by HSILA** [comment]
@mobassir94 The first issue I mentioned is a **representation issue**. This means that it will technically resume from 901/1000 but it will print out 1/1000 and it will be trained for 900 more epochs.

But the second issue is more critical. If you don't want to lose previously learned good weights, you should backup three **best_accuracy** files before resuming training, especially when your training loss has a lot of fluctuations.
